detector/detector.go

```diff
@@ -202,22 +136,42 @@ func (d *Detector) detect(creatorMethod string, issues *[]*issue.Issue) {
 	creator := reflect.ValueOf(d).MethodByName(creatorMethod)
 	detector := creator.Call([]reflect.Value{})[0]
 
-	if d.isSkip(reflect.Indirect(detector).FieldByName("DeepCheck").Bool(), reflect.Indirect(detector).FieldByName("Target").String()) {
+	if d.isSkip(
+		reflect.Indirect(detector).FieldByName("DeepCheck").Bool(),
+		reflect.Indirect(detector).FieldByName("TargetType").String(),
+		reflect.Indirect(detector).FieldByName("Target").String(),
+	) {
 		d.Logger.Info("skip this rule.")
 		return
 	}
-	if preprocess := detector.MethodByName("PreProcess"); preprocess.IsValid() {
-		preprocess.Call([]reflect.Value{})
+	if preProcess := detector.MethodByName("PreProcess"); preProcess.IsValid() {
+		preProcess.Call([]reflect.Value{})
 	}
 
-	for _, template := range d.Schema {
-		for _, resource := range template.Find("resource", reflect.Indirect(detector).FieldByName("Target").String()) {
-			detect := detector.MethodByName("Detect")
-			detect.Call([]reflect.Value{
-				reflect.ValueOf(resource),
-				reflect.ValueOf(issues),
-			})
+	switch reflect.Indirect(detector).FieldByName("TargetType").String() {
+	case "resource":
```

Comment:
__[gometalinter]__
* [goconst] 2 other occurrence(s) of "resource" found in: detector.go:227:7 test_helper.go:71:7

<!-- Comment by [SideCI](https://sideci.com) -->